samples/sample_definition_and_criteria.rst

```diff
@@ -39,36 +39,43 @@ Sample Criteria
         sample.kernel.cond_var:
           integration_platforms:
             - native_sim
-          tags: kernel condition_variables
+          tags:
+            - kernel
+            - condition_variables
           harness: console
           harness_config:
             type: one_line
             regex:
               - ".*Waited and joined with 3 threads"
 
-  **Guidelines for Samples:**
-    * Minimize the use of ``platform_allow`` and architecture filters.
-    * Do not mark the test as build only unless necessary.
-    * Any test case-specific configuration options are added using ``extra_args`` or
-      ``extra_configs`` options in the YAML file. The `prj.conf` file should have the in-common
-      configuration options.
-    * The sample itself can be evaluated using multiple configurations in the sample's YAML file.
-      For example, the sample is used to run with different logging modes using multiple scenarios
-      in ``samples/subsys/logging/syst``.
-    * Sample output can be validated leveraging the ``harness_config`` regex option,
-      wherever applicable.
-
-3. Samples should target multiple platforms and architectures.
-++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-  * Minimize the use of ``platform_allow`` and architecture filters as it limits the scope
-    of testing to the mentioned platforms and architectures.
-    Reference: :ref:`twister_script`
+
+  * Do not mark the test as build only. A sample should be able to build and
+    run on the documented platforms. Use the harness option to skip the
+    execution. Twister only attempts to flash and execute the build binary if
+    the harness is set to ztest or console.
+  * The default configuration for the sample should be in the ``prj.conf``
+    file and should be operational on all supported platforms. Do not rely on the
+    ``extra_args`` or ``extra_configs`` options in the YAML file to build the
+    sample.
+  * The tests should verify the default configuration of the sample and any
+    optional features documented in the sample's README file. Sample should
+    never be used to test functionality of the subsystem or module the sample
+    belongs to.
```

Comment:
> Sample should never be used to test functionality of the subsystem or module the sample belongs to.

Don't quite understand this, isn't the purpose of the various samples to test/show functionality of the subsystem, at least in networking samples do that.

Reply:
show functionality: yes, act as a test: no.
Obviously when you run a sample you test something, but this should not replace actual tests with full coverage under tests/.
In a sample, the only thing you test is buildability and in some cases if a sample performing as expected, i.e. you are testing the sample, not the subsystem.

Reply:
Sure, I agree fully. It was just not clear from the text what it actually means. Perhaps some of your clarification text could be added to the document.

Reply:
@jukkar  done